src/runtime/components/data/Table.vue

```diff
@@ -239,7 +239,7 @@ export default defineComponent({
           return
         }
 
-        onSelect(row)
+        attrs.onSelect ? onSelect(row) : selected.value.push(row)
```

Comment:
```suggestion
        attrs.onSelect ? attrs.onSelect(row) : selected.value.push(row)
```

Reply:
No need to check for `attrs.onSelect` twice.

Reply:
@benjamincanac Doesn't it make more sense to handle this logic here:

function onSelect (row) {
      if (!attrs.onSelect) {
+       selected.value.push(row)
-        return;
      }
      // @ts-ignore
      attrs.onSelect(row)
    }

Reply:
The `onSelect` method is called when clicking on the row, behaviour which should only occur when the `@select` event is specified. The bug in question is only when selecting all rows so I'd keep it inside the `selectAllRows` method.

Reply:
Thanks for the guidance! 
And I have a question is that this method is trigger only when no rows are selected. If the table not been selected all, it will clear the array, so I think it doesn't need a `forEach` to compare the data, right ? 
Maybe this might be another issue?

Reply:
> The `onSelect` method is called when clicking on the row, behaviour which should only occur when the `@select` event is specified. The bug in question is only when selecting all rows so I'd keep it inside the `selectAllRows` method.

Thanks for explaining 😊